### ISSUE TITLE
Fixie 3 fails test that passes in Fixie 2

### ISSUE THREAD

**Issue by jayotterbein** [fixie/fixie#259]
## Steps to Reproduce
I created a couple of test projects for this. They use the same library for code, aptly named "ClassLibrary1" and then two different solutions for testing this using Fixie 2.2.2 or Fixie 3.1.0.

Try running the test using both the Fixie2 and Fixie3 solutions.

[FixieHtmlTests.zip](https://github.com/fixie/fixie/files/6964506/FixieHtmlTests.zip)

The test currently fails using Fixie3 but passes with Fixie2.

**Comment by plioi** [comment]
Thanks for such a minimal repo. The behavior is, of course, the same for me as for you.

I took the relevant bits, then, and moved them into the Fixie repo locally, so that I could be debugging through your sample test *and* the Test Explorer "TestAdapter" code with its use of `AssemblyLoadContext`.

I was hoping that stepping through, I'd see calls entering the `AssemblyLoadContext` with a surprising (generated/dynamic) assembly name and that I'd then watch it fail to find it on disk or something similar. Surprisingly that never happened. Every invocation of the ALC was for unsurprising assembly names and the behavior seemed right for those.

I found that the code generated does live on disk, as well as a dll for it. They happen to wind up under `C:\Users\yourusername\AppData\Local\Temp\RazorEngine_randomcharactershere\ and within that we see a cs/dll/pdb triplet with names like: CompiledRazorTemplates.Dynamic.RazorEngine_e6a2c7eba66745c7b7f331bd1febbb44.cs`

The code generated is unsurprising:

```cs
// <auto-generated/>
#pragma warning disable 1591
namespace CompiledRazorTemplates.Dynamic
{
    #line hidden
    using System.Threading.Tasks;
    using System;
    using System.Collections.Generic;
    using System.Linq;
    internal class RazorEngine_e6a2c7eba66745c7b7f331bd1febbb44 : RazorEngine.Templating.TemplateBase<TheExampleTestProject.ViewModel>
    {
        #pragma warning disable 1998
        public async override global::System.Threading.Tasks.Task ExecuteAsync()
        {
            WriteLiteral("<div>");
            Write(Model.Value);
            WriteLiteral("</div>");
        }
        #pragma warning restore 1998
    }
}
#pragma warning restore 1591
```

Digging into the stack trace isn't helping. I think the root cause is far away from the moment where it results in an exception.

So, this is all very helpful confirmation that I'm not going to spot a quick win in my `AssemblyLoadContext` code. Since things actually run fine running with `dotnet fixie` at the command line, the only substantive difference here is that v3's Test Adapter loads the test assembly in-process using ALC and v2's instead kicks off the test assembly just like a `dotnet fixie` run.

In other words, the flows that work all just let the test project run as a normal console console app that is free to load all its dependencies itself through perfectly normal means. That's what happens for v2 from the console, v2 from Test Explorer, and v3 from the console.

The most likely solution (as I was speculating in our earlier discussion outside of this thread), will be for me to:

1. Using the v2 code as a guide, drop the use of the `AssemblyLoadContext` and reintroduce the older interprocess communication trickery that works.
2. Confirm that if I do that, the test starts passing.
3. If that all works, I'll need to further adjust the interprocess communication as the old way relied on a Windows-only named pipe mode ("Message" mode) and I think I will be able to use the cross-platform modes with a little extra care that messages arrive in full between the two processes.

**Comment by plioi** [comment]
Fixed by #260 and released in https://github.com/fixie/fixie/releases/tag/3.2.0
